src/Task.Util/Generator.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using CommandLine;
+using Microsoft.VisualStudio.Services.Agent;
+
+namespace Task.Util
+{
+    [Verb(Constants.TaskUtil.CommandLine.Commands.Generate)]
+    public class GenerateCommand
+    {
+        [Option(Constants.TaskUtil.CommandLine.Args.FolderPath, Required = true)]
+        public string FolderPath { get; set; }
+
+        [Option(Constants.TaskUtil.CommandLine.Args.TaskId)]
+        public string TaskId { get; set; }
+
+        [Option(Constants.TaskUtil.CommandLine.Args.TaskName)]
+        public string TaskName { get; set; }
+
+        [Option(Constants.TaskUtil.CommandLine.Flags.Overwrite)]
+        public bool Overwrite { get; set; }
+    }
+
+    public class Generator
+    {
+        public Generator(IHostContext hostContext)
+        {
+            _hostContext = hostContext;
+            _trace = _hostContext.GetTrace(nameof(Uploader));
+        }
+
+        public void GenerateTaskFiles(string folderPath, string taskId, string taskName, bool overwrite, object agentShutdownToken)
+        {
+            if (Directory.Exists(folderPath))
+            {
+                if (!overwrite)
+                {
+                    throw new Exception($"The folder {folderPath} already exists.");
+                }
+
+                Directory.Delete(folderPath, true);
+            }
+
+            Directory.CreateDirectory(folderPath);
+            Directory.SetCurrentDirectory(folderPath);
+
+            // Create task.json
+            taskId = String.IsNullOrEmpty(taskId) ? Guid.NewGuid().ToString() : taskId;
+            taskName = String.IsNullOrEmpty(taskName) ? "MyNewTask" : taskName;
+            string taskJson = 
```

Comment:
would it be worth it to look at pulling out these file content strings into template files?